### DISCUSSION TITLE
Some more features

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [a-r-j/graphein#145] [Q&A]
I want the volume, dipole moment, hydrogen bond donor, and hydrogen bond acceptor for the node feature in the protein residue graph. How can I do this using graphein ?

**Comment by a-r-j** [comment]
Hiya,

Could you clarify what you mean by volume?

Dipole moments aren’t currently implemented, but you can use a tool like [this](https://github.com/harmslab/pdbtools) to compute them.

For HBA/HBD, it should be pretty straightforward to define your own function based on this table:


Amino acids | Hydrogen donor atoms | Hydrogen acceptor atoms
-- | -- | --
Arginine (Arg, R) | NE, NH1 (2), NH2 (2) |  
Asparagine (Asn, N) | ND2 (2) | OD1 (2)
Aspartic acid (Asp, D) |   | OD1 (2), OD2 (2)
Glutamine (Gln, Q) | NE2 (2) | OE1 (2)
Glutamic acid (Glu, E) |   | OE1 (2), OE2 (2)
Histidine (His, H) | ND1, NE2 | ND1, NE2
Lysine (Lys, K) | NZ (3) |  
Serine (Ser, S) | OG | OG (2)
Threonine (Thr, T) | OG1 | OG1 (2)
Tryptophan (Trp, W) | NE1 |  
Tyrosine (Tyr, Y) | OH | OH

Eg.
```python
def hbond_type(n, d):
    If d[“residue_type”] == “Y”:
        d[“hbond_donor”] = True
        D[“hbond_acceptrot = True
    ….
```

**Reply by shrimonmuke0202** [reply]
Thanks for your reply,
Here volume means the volume of amino acid residue.

**Reply by a-r-j** [reply]
Graphein provides [featurisation with aaindex](https://graphein.ai/modules/graphein.protein.html#graphein.protein.features.nodes.aaindex.aaindex1) - make sure you install the extra dependency in `.requirements/extras.in`. You can see [here](https://www.genome.jp/aaindex/AAindex/list_of_indices) for the possible features - there are several volumes. There also seem to be some Hbond-related features you can look at.

**Reply by a-r-j** [reply]
Hey @shrimonmuke0202 I've added a Hbond donor and acceptor feature in #150 - What do you think? Is this what you had in mind?

**Reply by shrimonmuke0202** [reply]
Sorry for the late reply; this will help me a lot.

**Reply by a-r-j** [reply]
Great! Hbond acceptors/donors are now in `1.3.0`. Did you manage to get the volume features? I can also look into the dipole moments.

**Reply by shrimonmuke0202** [reply]
No, I did not get the volume features and the dipole moments.
